src/Nethermind/Nethermind.JsonRpc/JsonRpcProcessor.cs

```diff
@@ -96,6 +97,57 @@ private void BuildTraceJsonSerializer()
             return (token.ToObject<JsonRpcRequest>(_obsoleteBasicJsonSerializer), null);
         }
 
+        private (JsonRpcRequest Model, List<JsonRpcRequest> Collection) DeserializeObjectOrArray(string json)
+        {
+            void CreateRequest(JToken tokenElement, List<JsonRpcRequest> jsonRpcRequests)
+            {
+                UpdateParams(tokenElement);
+                JsonRpcRequest request = tokenElement.ToObject<JsonRpcRequest>(_obsoleteBasicJsonSerializer);
+                if (request != null)
+                {
+                    jsonRpcRequests.Add(request);
+                }
+            }
+
+            var parsedJson = JTokenUtils.ParseMulticontent(json);
+
+            List<JsonRpcRequest> collection = new();
```

Comment:
Even if we have only one result we're allocating List here.

Reply:
Yes.